src/api/custom-resolvers/mutations/delete-application-by-reference-number/index.ts

```diff
@@ -21,14 +22,11 @@ const deleteApplicationByReferenceNumber = async (
 
     const { referenceNumber } = variables;
 
-    const applications = (await context.db.Application.findMany({
-      where: {
-        referenceNumber: { equals: referenceNumber },
-      },
-    })) as Application;
+    // ids in application from provided application reference number
+    const application = await getApplicationByReferenceNumber(referenceNumber, context);
```

Comment:
Do you need to add `referenceNumber` validation before invoking this function?

Reply:
This isn't required since the GQL query has a required `referenceNumber` variable (defined in `type-defs`).

If this resolver is called without a `referenceNumber` it will error with e.g "required referenceNumber variable not provided"